File: programming 1/lab 6/src/main/java/server/commands/ShowCommand.java
package server.commands;


import common.communication.Response;
import server.managers.CollectionManager;

/**
 * Команда для вывода всех элементов коллекции.
 */
public class ShowCommand extends Command {

    private final CollectionManager cm;
    public ShowCommand(CollectionManager cm) {
        super("show", "вывести все элементы коллекции");
        this.cm = cm;
    }

    /**
     * Выполняет команду без аргументов.
     * @param arg аргумент команды (в данной команде не используется).
     */
    @Override
    public Response execute(String arg) {
        if (arg != null) throw new IllegalArgumentException(String.format("Команда %s не принимает аргументы\n", getName()));
        return cm.show();
    }
}

File: programming 1/lab 6 kotlin/kotlin/lab 6/src/main/java/server/commands/ShowCommand.kt
package server.commands

import common.communication.Response
import server.managers.CollectionManager


/**
 * Команда для вывода всех элементов коллекции.
 */
class ShowCommand(private val cm: CollectionManager) : Command("show", "вывести все элементы коллекции") {
    /**
     * Выполняет команду без аргументов.
     * @param arg аргумент команды (в данной команде не используется).
     */
    override fun execute(arg: String?): Response {
        require(arg == null) { "Команда $name не принимает аргументы" }
        return cm.show()
    }
}
